src/api/routes/application.js

```diff
@@ -12,14 +14,27 @@ export default (app) => {
      * Creates a new Company Application
      */
     router.post("/", validators.create, async (req, res, next) => {
-
         try {
+            await applicationMiddleware.exceededCreationTimeLimit(req.body.email);
+            await applicationMiddleware.deleteApplications(req.body.email);
+            const applicationService = new ApplicationService();
             // This is safe since the service is destructuring the passed object and the fields have been validated
-            const application = await (new ApplicationService()).create(req.body);
+            const application = await applicationService.create(req.body);
             return res.json(application);
         } catch (err) {
             console.error(err);
             return next(err);
         }
     });
```

Comment:
Validators are not called inside the route method, they are an argument, take a look at this example:

https://github.com/NIAEFEUP/nijobs-be/blob/2ddec2edbcc344c99c8c24592bc654e937e7ecc0/src/api/routes/offer.js#L115-L138

You might have to call as `(req) => applicationMiddleware.deleteApplications(req.body.email)`

Reply:
Also, why place the service in a variable that is only used once?